src/include/graph/on_disk_graph.h

```diff
@@ -30,12 +30,15 @@ class OnDiskGraphNbrScanState : public NbrScanState {
     OnDiskGraphNbrScanState(main::ClientContext* context, catalog::TableCatalogEntry* tableEntry,
         std::shared_ptr<binder::Expression> predicate);
     OnDiskGraphNbrScanState(main::ClientContext* context, catalog::TableCatalogEntry* tableEntry,
-        std::shared_ptr<binder::Expression> predicate, const std::string& propertyName,
+        std::shared_ptr<binder::Expression> predicate, std::vector<std::string> relProperties,
         bool randomLookup = false);
 
     Chunk getChunk() override {
-        return createChunk(currentIter->getNbrNodes(), currentIter->getEdges(),
-            currentIter->getSelVectorUnsafe(), propertyVector.get());
+        std::vector<common::ValueVector*> vectors;
+        for (auto& propertyVector : propertyVectors) {
+            vectors.push_back(propertyVector.get());
+        }
+        return createChunk(currentIter->getNbrNodes(), currentIter->getSelVectorUnsafe(), vectors);
```

Comment:
I have a question, are we creating a new chunk per scanFwd call? I have a feeling we might want to mark Chunk as EXPLICIT_COPY.